src/components/TodoProvider.jsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+import todoContext from "../todoContext";
+import { nanoid } from "nanoid";
+
+export default function TodoProvider({ children }) {
+  const [todos, setTodos] = useState([]);
+  const createNewTodo = (text) => {
+    setTodos([{ text, id: nanoid(), done: false }, ...todos]);
+  };
+  const completeTodo = (id) => {
+    const newTodos = todos.map((el) => {
+      if (el.id === id) {
+        return { ...el, done: !el.done };
+      } else {
+        return { ...el };
+      }
+    });
+    setTodos(newTodos);
+  };
+  const updateTodoText = (id, text) => {
+    const newTodos = todos.map((el) => {
+      if (el.id === id) {
+        return { ...el, text: text };
+      } else {
+        return { ...el };
+      }
+    });
+    setTodos(newTodos);
+  };
```

Comment:
Also here you can also use the ternary operator as well as the `text` shorthand to shorten this function. Check the suggestion: 
``` suggestion
    const updateTodoText = (id, text) => {
       const newTodos = todos.map((el) => el.id === id ? { ...el, text } : el);
       setTodos(newTodos);
};
```